jumanji/environments/combinatorial/routing/generation_methods_v1_uo.py

```diff
@@ -0,0 +1,572 @@
+import random
+from collections import deque
+from typing import List, Tuple, Union, Optional, Dict, Callable
+import numpy as np
+
+
+class Grid:
+    def __init__(self, rows: int, cols: int, grid: Optional[np.ndarray] = None) -> None:
+        """
+        Constructor for the Grid class.
+        Args:
+            rows (int): number of rows in the grid
```

Comment:
we can omit the type hints in the doctoring because we use type hints (they do the same in jumanji)